### PR TITLE
Build script for generating debrief-lite jar

### PR DESCRIPTION
**Pull request by ayeshare**
Supports ant build for debrief-lite

### CODE REVIEW

File: org.mwc.cmap.legacy/build.xml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+	<property name="bundleJavacSource" value="${javacSource}"/>
+	<property name="bundleJavacTarget" value="${javacTarget}"/>
+	<property name="bundleBootClasspath" value="${bootclasspath}"/>
+
+	<target name="init" depends="properties">
+		<condition property="pluginTemp" value="${buildTempFolder}/plugins"		>
+			<isset property="buildTempFolder"/>
+		</condition>
+		<property name="pluginTemp" value="${basedir}"/>
+		<condition property="build.result.folder" value="${pluginTemp}/org.mwc.cmap.legacy_1.0.255"		>
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: can we avoid having the hard-coded version number?

File: org.mwc.cmap.legacy/build.xml [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+			<isset property="buildTempFolder"/>
+		</condition>
+		<property name="build.result.folder" value="${basedir}"/>
+		<property name="temp.folder" value="${basedir}/temp.folder"/>
+		<property name="plugin.destination" value="${basedir}"/>
+		<condition property="p2.publish.parts" value="true"		>
+			<istrue value="${p2.gathering}"/>
+		</condition>
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: the next few lines look impressive, but I don't know what they're doing. Can you explain it?

File: org.mwc.cmap.legacy/build.xml [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	</target>
+
+	<target name="build.update.jar" depends="init" description="Build the plug-in: org.mwc.cmap.legacy for an update site.">
+		<delete dir="${temp.folder}"/>
+		<mkdir dir="${temp.folder}"/>
+		<antcall target="build.jars"/>
+		<antcall target="gather.bin.parts">
+			<param name="destination.temp.folder" value="${temp.folder}/"/>
+		</antcall>
+		<jar destfile="${plugin.destination}/org.mwc.cmap.legacy_1.0.255.jar" basedir="${temp.folder}/org.mwc.cmap.legacy_1.0.255" filesetmanifest="merge"/>
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: again - I'm nervous of the hard-coded versions here.

File: org.mwc.cmap.legacy/build.xml [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+				<exclude name="**/*.java"/>
+				<exclude name="**/package.htm*"/>
+			</fileset>
+		</copy>
+		<mkdir dir="${build.result.folder}"/>
+		<jar destfile="${build.result.folder}/cmap_legacy.jar" basedir="${temp.folder}/cmap_legacy.jar.bin"/>
+		<delete dir="${temp.folder}/cmap_legacy.jar.bin"/>
+	</target>
+
+	<target name="cmap_legacysrc.zip" depends="init" unless="cmap_legacysrc.zip">
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: hmm, we don't need to distribute zipped source code.  I think these steps can be removed.

File: org.mwc.cmap.legacy/build.xml [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+	<target name="build.jars" depends="init" description="Compile classes and build nested jars for the plug-in: org.mwc.cmap.legacy.">
+		<delete file="${compilation.problem.marker}" quiet="true"/>
+		<available property="cmap_legacy.jar" file="${build.result.folder}/cmap_legacy.jar"/>
+		<antcall target="cmap_legacy.jar"/>
+	</target>
+
+	<target name="checkCompilationResults" if="compilation.error.occured">
+		<echo file="${compilation.problem.marker}" message="org.mwc.cmap.legacy_1.0.255${line.separator}compilation.error.occured=${compilation.error.occured}"/>
+		<eclipse.logCompileError log="${compilation.prereq.log}" bundle="org.mwc.cmap.legacy_1.0.255">
+			<include name="C:/Users/Ayesha/git/debrief-dev/org.mwc.cmap.jfreechart/compilation.problem"/>
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: I'm sure we can't have hard-coded paths like this.

File: org.mwc.debrief.legacy/build.xml [hunk 6]
```diff
@@ -0,0 +1,312 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project name="org.mwc.debrief.legacy" default="build.jars" basedir=".">
+
+	<property name="p2.build.repo" value="file:${buildDirectory}/buildRepo"/>
+	<property name="basews" value="${ws}"/>
+	<property name="baseos" value="${os}"/>
+	<property name="basearch" value="${arch}"/>
+	<property name="basenl" value="${nl}"/>
+	<property name="bundleId" value="org.mwc.debrief.legacy"/>
+	<property name="bundleVersion" value="1.0.377"/>
```
Comment (IanMayo) [inline org.mwc.debrief.legacy/build.xml]: I guess this is the version you have to retrieve from Debrief build

File: org.mwc.debrief.legacy/build.xml [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+				<exclude name="**/*.java"/>
+				<exclude name="**/package.htm*"/>
+			</fileset>
+		</copy>
+		<jar destfile="${temp.folder}/debrief_legacy.jar" basedir="${temp.folder}/debrief_legacy.jar.bin">
+			<manifest>
+				<attribute name="Main-Class" value="Debrief.GUI.DebriefApp"/>
+				<attribute name="Specification-Title" value="Debrief Lite"/>
+				<attribute name="Specification-Version" value="1.0"/>
+				<attribute name="Specification-Vendor" value="MWC"/>
```
Comment (IanMayo) [inline org.mwc.debrief.legacy/build.xml]: Can we change the vendor to Deep Blue C Technology Ltd?

File: org.mwc.debrief.legacy/build.xml [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+				<attribute name="Implementation-Title" value="debriefLite"/>
+				<attribute name="Implementation-Version" value="0.1 ${TODAY}"/>
+				<attribute name="Implementation-Vendor" value="DeepBlue C Ltd"/>
+			</manifest>
+		</jar>
+		<delete dir="${temp.folder}/debrief_legacy.jar.bin"/>
+	</target>
+
+	<target name="debrief_legacysrc.zip" depends="init" unless="debrief_legacysrc.zip">
```
Comment (IanMayo) [inline org.mwc.debrief.legacy/build.xml]: again, we don't need to collate/distribute the sources.

File: org.mwc.cmap.legacy/build.xml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+		</condition>
+		<property name="pluginTemp" value="${basedir}"/>
+		<property name="temp.folder" value="${basedir}/temp.folder"/>
+	</target>
+
+	<target name="cmap_legacy.jar" depends="init" unless="cmap_legacy.jar" description="Create jar: org.mwc.cmap.legacy cmap_legacy.jar.">
+		<delete dir="${temp.folder}/cmap_legacy.jar.bin"/>
+		<mkdir dir="${temp.folder}/cmap_legacy.jar.bin"/>
+		<path id="cmap_legacy.jar.classpath">
+			<pathelement path="../org.mwc.cmap.jfreechart/libs/jcommon-1.0.23.jar"/>
```
Comment (IanMayo) [inline org.mwc.cmap.legacy/build.xml]: @ayeshare - is it possible to use a wildcard for these jars?  

If we're pulling in all the jars from the folder then using a wildcard will make the build less fragile if a .jar version changes.

File: org.mwc.debrief.legacy/build.xml [hunk 10]
```diff
@@ -23,17 +22,8 @@
...
-		<taskdef name="xmltask" classname="com.oopsconsultancy.xmltask.ant.XmlTask"
-			classpath="conf/xmltask.jar"
-			onerror="report"/>
-		<xmltask source="../org.mwc.debrief.product/debriefng.product">
-				<copy path="/product[@attr='version']" property="product.version"/>
-		</xmltask>
-		<echo>product version is: ${product.version}</echo>
-	</target>
-	
+	<xmlproperty file="pom.xml"/>
```
Comment (IanMayo) [inline org.mwc.debrief.legacy/build.xml]: Wow, that's an impressive Ant command - hadn't seen that before!

File: org.mwc.cmap.legacy/build.xml (hunk 9 above)
Comment (ayeshare) [reply org.mwc.cmap.legacy/build.xml]: made this change

**Pull request merged**